### ISSUE TITLE
Setting codec should override default additional_codecs

### ISSUE THREAD

**Issue by russorat** [logstash-plugins/logstash-input-http#23]
It seems like setting json or json_lines for the codec should override the defaults set by additional_codecs.

For example: 
```
http {
    host => "0.0.0.0"
    port => 7070
    threads => 4
    codec => 'json_lines'
  }
```
If someone makes a post with the content-type: application/json, the code uses the json codec instead of the json_lines codec because it is overwritten by the additional_codecs default option.

This change would make the codec option override the default for additional_codecs.

**Comment by jsvd** [comment]
The idea behind "codec" vs "additional_codec" interplay is that the plugin checks the additional codecs first and then falls back to codec if no match is found.
Making codec override additional_codec would make the latter never be used, since "codec" is *Always* defined (default is plain).
My suggestion for your use case would be to just set codec to json_lines AND also 
`additional_codecs => { }`

**Comment by russorat** [comment]
i guess it was strange to me because the name "additional codecs" implies that the other codec field is the "primary codec". So if i set the primary without setting the additional, i would not expect the additional to be used first.

Your workaround does indeed work, but it should be made clear somewhere that the additional codecs list is checked first, and the codec field is used as a fallback. The only way to figure that out today is by reading the source code.

https://www.elastic.co/guide/en/logstash/current/plugins-inputs-http.html#plugins-inputs-http-additional_codecs

**Comment by jsvd** [comment]
Agreed, I suck at naming :( I'll add that documentation..thanks for reporting this

**Comment by jsvd** [comment]
version 1.0.3 has been released with #26, which addresses this with better documentation
